frontend/src/hooks/useTimeout.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useRef } from 'react'
+
+export const useTimeout = (
+  callback: (...args: unknown[]) => void,
+  delay: number | null,
+) => {
+  const savedCallback = useRef(callback)
+
+  useEffect(() => {
+    if (delay == null) return
+
+    let timeoutId: number | null = null
+
+    timeoutId = window.setTimeout(() => {
+      savedCallback.current()
+    }, delay)
+
+    return () => {
+      if (timeoutId) {
+        window.clearTimeout(timeoutId)
+      }
+    }
```

Comment:
should be fine if `clearTimeout` returns `undefined`; will try.

Reply:
Based on the discussion we had yesterday, if react invokes the cleanup function only once, my original comment holds, and you can remove the test entirely. Since `timeoutId` will always be defined, the cleanup can simply be:
```javascript
return () => window.clearTimeout(timeoutId)
```